### PR TITLE
Improve select performance by keeping split classes

### PR DESCRIPTION
**Pull request by crazyjul**
Optimize class parsing and selector matching

### CODE REVIEW

File: src/css_selector.h [hunk 1]
```diff
@@ -144,9 +145,18 @@ namespace litehtml
 		css_attribute_selector(const css_attribute_selector& val)
 		{
 			this->val	= val.val;
+			class_val	= val.class_val;
 			condition	= val.condition;
 			attribute	= val.attribute;
 		}
+
+		css_attribute_selector(css_attribute_selector&& val) /*noexcept*/
```
Comment (NuclearCookie) [inline src/css_selector.h]: why the commented noexcept?

Comment (crazyjul) [inline src/css_selector.h]: It should be added in the future, but not supported by VS2012

Comment (NuclearCookie) [inline src/css_selector.h]: :+1:

Comment (NuclearCookie) [inline src/css_selector.h]: could be defined in os_types ( like we do with override and final )

File: src/html_tag.cpp [hunk 2]
```diff
@@ -2368,39 +2369,48 @@ bool litehtml::html_tag::set_pseudo_class( const tchar_t* pclass, bool add )
 
 bool litehtml::html_tag::set_class( const tchar_t* pclass, bool add )
```
Comment (NuclearCookie) [inline src/html_tag.cpp:2370]: implementation does not really match function name. This should be "add_class"

File: src/html_tag.cpp [hunk 3]
```diff
@@ -2368,39 +2369,48 @@ bool litehtml::html_tag::set_pseudo_class( const tchar_t* pclass, bool add )
...
+				m_class_values.push_back( std::move( _class ) );
+				changed = true;
+			} 
 		}
 	} else
 	{
-		auto end = std::remove(tokens.begin(), tokens.end(), pclass);
-
-		if(end == tokens.end())
+		for( auto & _class : classes )
```
Comment (NuclearCookie) [inline src/html_tag.cpp]: can be const &

File: src/html_tag.cpp (hunk 2 above)
Comment (crazyjul) [inline src/html_tag.cpp:2370]: function above set_pseudo_class has similar signature

**Pull request merged**